src/interceptors/exception.interceptor.ts

```diff
@@ -0,0 +1,35 @@
+import {
+  ArgumentsHost,
+  Catch,
+  ExceptionFilter,
+  HttpException,
+  Logger,
+} from '@nestjs/common';
+import { Response } from 'express';
+import { APIResponse } from 'src/utils/apiResponse.utils';
+
+@Catch(HttpException)
+export class HttpExceptionFilter implements ExceptionFilter {
+  private logger = new Logger();
+
+  catch(exception: HttpException, host: ArgumentsHost) {
+    let err = exception['response'];
+    err = err.response ? err.response : JSON.stringify(err);
+    const context = host.switchToHttp();
+    const response = context.getResponse<Response>();
+    const status = exception.getStatus();
+    this.logger.error(
+      `Message : ${err}`,
+      `${context.getRequest().method} ${context.getRequest().url}`,
+    );
+    const error =
+      typeof exception.getResponse() === 'string'
+        ? exception.getResponse()
+        : exception.getResponse() &&
+            typeof exception.getResponse()['message'] === 'object'
+          ? exception.getResponse()['message'].join(', ')
+          : exception.getResponse()['message'];
```

Comment:
PullProofAI

- Redundant code for extracting error messages in the `HttpExceptionFilter`.
+ Simplify the error message extraction logic to improve readability and maintainability.
```

#### Explanation:
1. The code snippet from line 16 to 31 contains redundant logic for extracting error messages from exceptions.
2. Simplifying this logic will enhance code readability and maintainability by reducing unnecessary complexity.

These improvements will help in enhancing the error handling capabilities of the `HttpExceptionFilter` interceptor and make the code more maintainable in the long run.


<!-- This is an auto-generated comment by PullProofAI -->